commands/rank.js

```diff
@@ -0,0 +1,141 @@
+const Discord = require('discord.js');
+const Enmap = require("enmap");
+
+const ALLOWED_RANKS = [
+  "Iron1",
+  "Iron2",
+  "Iron3",
+  "Bronze1",
+  "Bronze2",
+  "Bronze3",
+  "Silver1",
+  "Silver2",
+  "Silver3",
+  "Gold1",
+  "Gold2",
+  "Gold3",
+  "Platinum1",
+  "Platinum2",
+  "Platinum3",
+  "Diamond1",
+  "Diamond2",
+  "Diamond3",
+  "Immortal1",
+  "Immortal2",
+  "Immortal3",
+  "Radiant"
+];
+
+exports.run = async (client, message, args, level) => {
+  client.ranks = new Enmap({name: "ranks"});
+  const channel = message.channel;
+
+  const subcommand = args[0];
+
+  if(!subcommand) {
+    return sendHelp(channel);
+  }
+
+  switch(subcommand) {
+    case "set":
+      const rank = args[1];
+
+      if(!rank || !ALLOWED_RANKS.includes(rank)) {
+        // invalid rank, send allowed ranks
+        return sendRanks(channel);
+      } else {
+        const key = message.author.id;
+
+        // save rank
+        client.ranks.set(key, {
+          userId: key,
+          rank: rank
+        });
+
+        channel.send(message.author.username + "'s rank set to: " + rank);
+      }
+
+      break;
+    case "get":
+      const userArg = args[1];
+
+      if(!userArg || !userArg.startsWith("<@")) {
+        // missing/wrong argument, send help
+        return sendHelp(channel);
+      }
+
+      // extract userId
+      const userId = userArg.replace(/[<@!>]/g, '');
+
+       // retrieve data for user
+      const result = client.ranks.get(userId);
+
+      if(result) {
+
+        channel.send(getUsername(client, userId) + " is: " + result.rank);
+      } else {
+        channel.send("User not found");
+      }
+
+      break;
+    case "leaderboard":
+      // retrieve all data
+      const ranks = client.ranks.array();
+
+      // sort
+      const sortedRanks  = ranks.sort((a, b) => getRankIndex(b.rank) - getRankIndex(a.rank));
```

Comment:
Maybe one thing that would be nice is if we could store it sorted so we dont have to sort on retrieval.

Reply:
is that possible with enmap?

Reply:
Is this where you based your example from? https://gist.github.com/eslachance/12e2239aa353b350e075a4b006238335